packages/engine/lib/error/src/report.rs

```diff
@@ -0,0 +1,315 @@
+    pub fn compat(self) -> Report {
+        Report {
+            inner: self.inner,
+            _context: PhantomData,
+        }
+    }
+
+    /// Returns the backtrace of the error, if captured.
+    ///
+    /// Note, that `RUST_BACKTRACE` or `RUST_LIB_BACKTRACE` has to be set to enable backtraces.
+    ///
+    /// [`ReportBackTrace`]: crate::tags::ReportBackTrace
+    #[must_use]
+    #[cfg(feature = "backtrace")]
+    pub fn backtrace(&self) -> Option<&Backtrace> {
+        let backtrace = self.inner.backtrace.as_ref().unwrap_or_else(|| {
+            // Should never panic as it's either stored inside of `Report` or is provided by a frame
+            self.request::<tags::ReportBackTrace>()
+                .next()
+                .expect("Backtrace is not available")
+        });
+        if backtrace.status() == BacktraceStatus::Captured {
+            Some(backtrace)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the span trace of the error, if captured.
+    ///
+    /// Note, that [`ErrorLayer`] has to be enabled to enable span traces.
+    ///
+    /// [`ReportSpanTrace`]: crate::tags::ReportSpanTrace
+    /// [`ErrorLayer`]: tracing_error::ErrorLayer
+    #[must_use]
+    #[cfg(feature = "spantrace")]
+    pub fn span_trace(&self) -> Option<&SpanTrace> {
+        let span_trace = self.inner.span_trace.as_ref().unwrap_or_else(|| {
+            // Should never panic as it's either stored inside of `Report` or is provided by a frame
+            self.request::<tags::ReportSpanTrace>()
+                .next()
+                .expect("SpanTrace is not available")
+        });
+        if span_trace.status() == SpanTraceStatus::CAPTURED {
+            Some(span_trace)
+        } else {
+            None
+        }
+    }
+
+    /// Returns an iterator over the [`Frame`] stack of the report.
+    pub const fn frames(&self) -> Frames<'_> {
+        Frames::new(self)
+    }
+
+    /// Creates an iterator over the [`Frame`] stack requesting the type specified by [`I::Type`].
+    ///
+    /// [`I::Type`]: provider::TypeTag::Type
+    pub fn request<'p, I: 'static>(&'p self) -> Requests<'p, I>
+    where
+        I: TypeTag<'p>,
+    {
+        Requests::new(self)
+    }
+
+    /// Creates an iterator over the [`Frame`] stack requesting a reference of type `T`.
+    pub const fn request_ref<T: ?Sized + 'static>(&self) -> Requests<'_, Ref<T>> {
+        Requests::new(self)
+    }
+
+    /// Creates an iterator over the [`Frame`] stack requesting a value of type `T`.
+    pub const fn request_value<T: 'static>(&self) -> Requests<'_, Value<T>> {
+        Requests::new(self)
+    }
+
+    /// Returns if `E` is the type held by any frame inside of the report.
+    #[must_use]
+    pub fn contains<E>(&self) -> bool
+    where
+        E: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static,
+    {
+        self.frames().any(Frame::is::<E>)
+    }
+
+    /// Downcast this error object by a shared reference.
```

Comment:
Perhaps clarify that this returns the _first_ error of that type

Reply:
Very good point!